File: Python/Sum of two integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask = 0xFFFFFFFF
        boundary = 0x7FFFFFFF

        while (b & mask) != 0:
            bitsum = a ^ b & mask
            carry = a & b
            shifted_carry = carry << 1
            a = bitsum
            b = shifted_carry & mask

        if (a > boundary):
            return ~(a ^ mask)
        return a

print(Solution.getSum(Solution, 15, -4))

File: C++/Sum of two integers.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int getSum(int a, int b) {
        /*if (b == 0) {
            return a;
        }
        int sum = a ^ b;
        int carry = a & b;
        int shifted_carry = carry << 1;
        return(getSum(sum, shifted_carry));*/

        while (b != 0) {
            int sum = a ^ b;
            int carry = a & b;
            int shifted_carry = carry << 1;
            a = sum; b = shifted_carry;
        }
        return a;
    }
};

int main() {
    Solution problem;
    int a, b;
    cin >> a;
    cin >> b;
    cout << problem.getSum(a,b);
}